### ISSUE TITLE
Failed to turn off `Restricted Access` for Documentation Plugin

### ISSUE THREAD

**Issue by martin-luo** [strapi/strapi#22295]
### Required System information

- Node.js version: 20.10.0
- NPM version: 10.5.0
- Strapi version: 5.4.2
- Database:Postgresql
- Operating system: macOS 15.1.1 (24B91)
- Is your project Javascript or Typescript: TypeScript

### Steps to reproduce the behavior

1. Go to http://localhost:1337/admin/settings/documentation
2. Click on `ON` in `Settings -> Restricted Access`
3. Type in a valid password
4. Click 'Save'
5. Click on `OFF` in `Settings -> Restricted Access`
6. Click 'Save'
7. See error

### Expected behavior

Restricted Access is turned off and user can visit documentations, e.g. http://localhost:1337/documentation/v1.0.0 without entering the password

### Screenshots
![Documentation setting page with error](https://github.com/user-attachments/assets/1014b958-859a-4152-b9b3-385aa1d5444b)
![Documentation setting page with error and network tab details](https://github.com/user-attachments/assets/76177f96-37ad-4dfb-a74c-66200f290e0c)

**Comment by cwilso03** [comment]
A workaround is to manually (through curl, postman, etc.) submit the `PUT` request to update the settings without including the empty `password` property, like so:

```
PUT http://localhost:1337/documentation/updateSettings
Content-Type: application/json
Authorization: bearer <token value>

{
  "restrictedAccess": false
}
```

To get the JWT, send a `POST` to the admin login URL:

```
POST http://localhost:1337/admin/login
Content-Type: application/json

{
  "email": "youremail@yourdomain.com"
  "password": "yourpassword"
}
```

The JWT is in the response body's `data.token` property.

**Comment by jigyarth-searce** [comment]
Yes I also faced this issue. My workaround was to use a chrome extension and intercept the request, inserting password to the request body. I used a chrome extension `[Temper dev] for intercepting the request(https://chromewebstore.google.com/detail/tamper-dev/mdemppnhjflbejfbnlddahjbpdbeejnn)`

Your new curl should be 
`curl 'http://localhost:1337/documentation/updateSettings' \
  -X 'PUT' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8,gu;q=0.7' \
  -H 'Connection: keep-alive' \
  -H 'Cookie: twk_uuid_60df10bf7f4b000ac03ab6a8=%7B%22uuid%22%3A%221.HQ1Exf1MVTTpBYKl2iBWr2J0AixYUuQ6EaaXAEapfhhPrvw0MLLkAHZkPSKpyd3QYo6HgsUDZ4OeXlS919v8aWOPtJ3eKdLpLfMm9%22%2C%22version%22%3A3%2C%22domain%22%3Anull%2C%22ts%22%3A1728545072296%7D' \
  -H 'Origin: http://localhost:1337' \
  -H 'Referer: http://localhost:1337/admin/settings/documentation' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/132.0.0.0 Safari/537.36' \
  -H 'accept: application/json' \
  -H 'authorization: Bearer <YOUR JWT TOKEN>' \
  -H 'content-type: application/json' \
  -H 'sec-ch-ua: "Not A(Brand";v="8", "Chromium";v="132", "Google Chrome";v="132"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"restrictedAccess":false,"password":"<ENTER-YOUR-PASSWORD>"}'`

**Comment by dansp89** [comment]
My solution with middleware

```typescript
// ./src/middlewares/documentation-settings.ts
export default () => async (ctx, next) => {
    if (ctx.method === 'PUT' && ctx.url === '/documentation/updateSettings') {
        const body = ctx.request.body;
        if (body && body.restrictedAccess === false) delete body.password;

        ctx.request.body = body;
        console.log('[middleware][PUT]::documentation-settings', { body });
    }
    
    await next();
};
````

```typescript
// ./config/middlewares.ts
export default [
  'strapi::logger',
  'strapi::errors',
  'strapi::security',
  'strapi::cors',
  'strapi::poweredBy',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  // Add here
  'global::documentation-settings'
];
````
